numba/cuda/cudadrv/devicearray.py

```diff
@@ -111,6 +112,24 @@ def __init__(self, shape, strides, dtype, stream=0, gpu_data=None):
         self.gpu_data = gpu_data
         self.stream = stream
 
+    @devices.require_context
+    def __getitem__(self, item):
+        return self._do_getitem(item)
+
+    def getitem(self, item, stream=0):
```

Comment:
If `__getitem__` needs to be decorated with `@devices.require_context`, then this ought to have it as well (this issue was pre-existing, not introduced by this PR, but it would be good to fix now whilst modifying this code).

For a little background - I'm not entirely clear about the circumstances under which `devices.require_context` is needed - I do wonder if it could be elided here, given that a context would have been needed to allocate an array in the first place... It may be that it ensures a context usable by Numba is available in the event that a call to another CUDA-using library changed the context in the interim period between array creation and the call to this method.